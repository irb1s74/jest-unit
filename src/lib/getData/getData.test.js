import axios from "axios";
import {getData} from "./getData";
import resUsers from './users.json'

jest.mock('axios');

describe('getData', () => {
    let response = {
        data: resUsers
    };
    beforeEach(() => {
        response = {
            data: resUsers
        };
    })

    test('validate data', async () => {
        axios.get.mockReturnValue(response);
        const data = await getData();
        expect(axios.get).toBeCalledTimes(1);
        expect(data).toEqual(['1', '2', '3'])
        expect(data).toMatchSnapshot()
    })
})
