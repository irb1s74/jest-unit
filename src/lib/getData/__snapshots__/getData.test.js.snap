// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData validate data 1`] = `
Array [
  "1",
  "2",
  "3",
]
`;
